### PR TITLE
Rename `UiVerbosity` to `UiLayout` and make its variants more precise

### PR DESCRIPTION
**Pull request by abey79**
### What

* part of #6245 

Pure refactor. The second commit clarify the meaning of the various `UiLayout` variants. The other commits are just renames.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6291?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6291?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6291)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by jleibs** [comment]
`UiContext` seems too vague of a term... there's a lot of other ui-related context in other places such as `ViewerContext::re_ui` or `egui::Context`, and so just based on the name it's not immediately obvious to me that this context is just about density/verbosity of presentation.

A name more like `RequestedLayout` would make more sense to me.

**Comment by abey79** [comment]
I agree. `UiLayout` maybe? Slightly shorter and better refers to both what's being requested and the "destination context" in which the ui will appear.

File: crates/re_viewer_context/src/component_ui_registry.rs [hunk 1]
```diff
@@ -7,36 +7,40 @@ use re_types::ComponentName;
...
-/// Controls how mich space we use to show the data in a component ui.
+/// Specifies the context in which the UI is used and the constraints it should follow.
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum UiVerbosity {
-    /// Keep it small enough to fit on one row.
-    Small,
+pub enum UiLayout {
+    /// Display a short summary. Used in lists.
+    ///
+    /// Keep it small enough to fit on half a row. Text should truncate.
```
Comment (emilk) [inline crates/re_viewer_context/src/component_ui_registry.rs:15]: "half a row" ? O.o

**Review by emilk — APPROVED** [review]

File: crates/re_viewer_context/src/component_ui_registry.rs (hunk 1 above)
Comment (abey79) [reply crates/re_viewer_context/src/component_ui_registry.rs:15]: I refer to the list item's second column here. I'll clarify the comment.

**Pull request merged**